container/go/pkg/oci/write_test.go

```diff
@@ -0,0 +1,157 @@
+// Copyright 2015 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//////////////////////////////////////////////////////////////////////
+// Tests for Write function.
+
+package oci
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+	"testing"
+
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/layout"
+	"github.com/google/go-containerregistry/pkg/v1/tarball"
+	"github.com/google/go-containerregistry/pkg/v1/validate"
+	"github.com/pkg/errors"
+)
+
+var (
+	blob     = "blobs/"
+	testPath = "testdata/test_write1"
+)
+
+var testCases = []struct {
+	// name is the name of the test case.
+	name string
+	// dataPath is where the test data is found.
+	dataPath string
+	// path to the image manifest relative to the dataPath.
+	manifestPath string
+	// content is the OCI layout content where the key is the file path and value is the sha256 sum.
```

Comment:
I would simplify this comment further to
```
// content is a mapping from path to a file expected to exist in the image relative to "dataPath" to
// the expected sha256 digest of that file.
```

This clearly states what the value of the key & data portions of this map are. The OCI bit is an implementation detail. I would mention the "OCI" part in the comments section for each testCase value below if you want.